.github/workflows/cortex_m_virtual_hardware.yml

```diff
@@ -7,74 +7,123 @@
 name: Cortex-M on Arm Virtual Hardware
 
 # https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onschedule
-on:
-  schedule:
-    - cron: '0 4 * * *'
+on: [push, pull_request]
```

Comment:
As written, this will run on every push to any branch and on every PR event. I'm going to revise this to a more narrow subset. I'm thinking ci:run_full and postmerge makes sense for this.

Reply:
Mans: this could probably be run pre-merge.

Reply:
@MatthiasHertel80 - I can't modify your main branch, so I can't update this myself. I'm going to suggest this:

```
on:
  # Allow manually triggering of the workflow.
  workflow_dispatch: {}

  workflow_call:
    inputs:
      trigger-sha:
        required: true
        type: string
    secrets:
      tflm-bot-token:
        required: true
```

I want to test it out in CI first, then start requiring it for PRs.